code/modules/reagents/chemistry/reagents/alchemical_potions.dm

```diff
@@ -0,0 +1,224 @@
+
+
+/datum/reagent/magic/teleportarium/unstable
+	name = "unstable teleportarium"
+	description = "Transforms your target into a harmless animal."
+
+/datum/reagent/magic/teleportarium/unstable/teleport_sucker(mob/living/L)
+	do_teleport(L, get_turf(L), tele_range, asoundin = 'sound/effects/phasein.ogg', channel = TELEPORT_CHANNEL_BLUESPACE)
+
+//	----	INVISIBILIUM	----
+
+/datum/reagent/magic/invisibilium
+	name = "invisibilium"
+	description = "Transforms your target into a harmless animal."
+	color = "#cccccc"
+	taste_description = "something that makes you angry!"
+	metabolization_rate = 0.5 * REAGENTS_METABOLISM
+	var/original_alpha = 1
+	var/charge = 0
+	var/last_volume = 0
+
+/datum/reagent/magic/invisibilium/on_mob_metabolize(mob/living/L)
+	. = ..()
+	original_alpha = L.alpha
+
+/datum/reagent/magic/invisibilium/on_mob_life(mob/living/L)
+	. = ..()
+	if (volume >= last_volume)
+		last_volume = volume
+		charge = max(300,charge + volume * 10)
+	else
+		charge = volume/last_volume * 300
+	animate(L,alpha = CLAMP(255 - charge,0,255),time = 10)
+
+/datum/reagent/magic/invisibilium/on_mob_end_metabolize(mob/living/L)
+	..()
+	L.alpha = original_alpha
+
+//	----	LEVITATIUM	----
+
+/datum/reagent/magic/levitatium
+	name = "polymorphine"
+	description = "Transforms your target into a harmless animal."
+	color = "#cccccc"
+	taste_description = "something that makes you angry!"
+	metabolization_rate = 3 * REAGENTS_METABOLISM
+
+/datum/reagent/magic/levitatium/on_mob_metabolize(mob/living/L)
+	..()
+	L.add_movespeed_modifier(MOVESPEED_ID_LEVITATION_POTION, priority=13, multiplicative_slowdown=-2, movetypes=(FLYING|FLOATING))
+
+/datum/reagent/magic/levitatium/on_mob_end_metabolize(mob/living/L)
+	..()
+	L.remove_movespeed_modifier(MOVESPEED_ID_LEVITATION_POTION)
+
+//	----	DRAUGHT OF MIDAS	----
+
+
+/datum/reagent/magic/midas
+	name = "draught of midas"
+	description = "Transforms your target into a harmless animal."
+	color = "#cccccc"
+	taste_description = "something that makes you angry!"
+	var/static/c_v = 10
+
+/datum/reagent/magic/midas/reaction_obj(obj/O, reac_volume)
+	if(ismob(O.loc)) //handled in human acid_act()
+		return
+	reac_volume = round(reac_volume,0.1)
+	if(istype(O, /obj/machinery/door) || (O.type!=/obj/structure/mineral_door/gold && istype(O, /obj/structure/mineral_door)))
+		if (reac_volume<c_v)
+			return
+		var/obj/structure/mineral_door/D = new /obj/structure/mineral_door/gold(get_turf(O))
+		qdel(O)
+		D.Open()
+	else if (reac_volume>=c_v/2)
+		if (O.type == /obj/item/bikehorn)	//bikehorn
+			new /obj/item/bikehorn/golden(get_turf(O))
+		else if (O.type == /obj/item/instrument/violin)
+			new /obj/item/instrument/violin/golden(get_turf(O))
+		else if (istype(O,/obj/item/stack/sheet/mineral) && O.type!=/obj/item/stack/sheet/mineral/gold)
+			var/obj/item/stack/sheet/mineral/catalyst = O
+			var/obj/item/stack/sheet/mineral/gold/output = new /obj/item/stack/sheet/mineral/gold(get_turf(O))
+			output.add(catalyst.amount-1)
```

Comment:
You can pass in the amount you want in the new of the sheet
`/obj/item/stack/Initialize(mapload, new_amount, merge = TRUE)`
So, this would just become
```
			var/obj/item/stack/sheet/mineral/catalyst = O
			 new /obj/item/stack/sheet/mineral/gold(get_turf(O), catalyst.amount)
```

Reply:
Thanks I didn't know that.